sim/warrior/charge.go

```diff
@@ -6,28 +6,82 @@ import (
 	"github.com/wowsims/cata/sim/core"
 )
 
+func performCharge(sim *core.Simulation, warrior *Warrior, metrics *core.ResourceMetrics, rage float64) {
+	warrior.AddRage(sim, rage, metrics)
+	if warrior.DistanceFromTarget > core.MaxMeleeRange {
+		warrior.PseudoStats.MovementSpeedMultiplier *= 2
```

Comment:
I would assume that it's much more than a 2x speed increase? The way we measured this for Feral Charge is by logging dummy tests from variable target distances, where the tester mashed a melee range special after pressing Charge. We then looked at the delay between the Charge cast and the special cast in logs across the data set to fit a linear correlation versus distance.

Reply:
I did some testing in game with logging turned on and it was very consistent with 3x so I changed it to that.
It's *almost* instant in the timeline, like 0.2-0.3s to charge back in.